qiskit/transpiler/passes/layout/vf2_layout.py

```diff
@@ -129,6 +130,34 @@ def run(self, dag):
         if self.coupling_map is None:
             raise TranspilerError("coupling_map or target must be specified.")
         self.avg_error_map = self.property_set["vf2_avg_error_map"]
+        # Run rust fast path if we have a target and no randomization
+        if self.seed == -1 and self.target is not None:
+            try:
+                layout = vf2_layout_pass(
+                    dag,
+                    self.target,
+                    self.strict_direction,
+                    self.call_limit,
+                    self.time_limit,
+                    self.max_trials,
+                    self.avg_error_map,
+                )
+            except MultiQEncountered:
+                self.property_set["VF2Layout_stop_reason"] = VF2LayoutStopReason.MORE_THAN_2Q
+                return
+            if layout is None:
+                self.property_set["VF2Layout_stop_reason"] = VF2LayoutStopReason.NO_SOLUTION_FOUND
+                return
+
+            self.property_set["VF2Layout_stop_reason"] = VF2LayoutStopReason.SOLUTION_FOUND
+            mapping = {dag.qubits[virt]: phys for virt, phys in layout.items()}
+            chosen_layout = Layout(mapping)
+            self.property_set["layout"] = chosen_layout
+            for reg in dag.qregs.values():
+                self.property_set["layout"].add_register(reg)
+            return
+        # We can't use the rust fast path because we have a seed set, or no target so continue with
+        # the python path
```

Comment:
I feel like this is actually possible certainly for the seed, and (with greater annoyance) also for the coupling map, but I wouldn't necessarily bother in this PR now.  It'd be good to remove all this extra Python code at some point, though.

Reply:
TBH, I was thinking instead it would be better to deprecate these options on the pass and support target only moving forward and no randomization moving forward. For `VF2Layout` I could see the argument for maybe still supporting coupling map but for `VF2PostLayout` it doesn't really make any sense.

I only wanted to concentrate on the common path that we use in the preset pass managers for this PR though.

Reply:
Yeah, I generally agree.  My main concern is getting rid of the extra Python code, mostly - it's unpleasant that we're maintaining two separate paths through VF2.  I don't feel like we need to do anything about it right now.